rclpy/test/test_clock.py

```diff
@@ -271,6 +293,18 @@ def test_sleep_until_time_in_past(default_context, clock_type):
     assert stop - start < a_small_amount_of_time
 
 
+@pytest.mark.parametrize(
+    'clock_type', (ClockType.SYSTEM_TIME, ClockType.STEADY_TIME, ClockType.ROS_TIME))
+def test_sleep_for_negative_duration(default_context, clock_type):
+    clock = Clock(clock_type=clock_type)
+    sleep_duration = Duration(seconds=-1)
+    start = clock.now()
+    assert clock.sleep_for(sleep_duration)
+    stop = clock.now()
+    a_small_amount_of_time = Duration(seconds=0.01)
+    assert stop - start < a_small_amount_of_time
```

Comment:
This could cause some flakyness issues in CI.
I wouldn't expect much problem if we use a bigger tolerance, e.g. 0.5.

Reply:
Using 0.5s in 3dd61e7